### DISCUSSION TITLE
Breakpointing in dev mode causes computers to stop working

### DISCUSSION THREAD

**Discussion by uecasm** [cc-tweaked/CC-Tweaked#1091] [Q&A]
### Minecraft Version

1.16.x

### Version

1.100.4

### Details

In a mod development environment, with a peripheral method tagged with `@LuaFunction(mainThread = true)`, placing a breakpoint inside the method and quickly stepping through works ok, but if left stopped at the breakpoint for several seconds, after resuming execution the method completes but its output is not returned to the lua code.

Running a second such method after this timeout occurs locks up the lua interpreter without executing the peripheral method; it does still respond to Ctrl-T/Ctrl-R etc and standard methods continue to work, but there does not appear to be any way to allow it to call mainThread methods again without restarting the entire Minecraft server.

**Comment by SquidDev** [comment]
CC attempts to limit the amount of work a computer can do on the main thread each server tick, giving each computer a budget of 5ms/tick and a global budget of 10ms/tick. When you break within a method, you're obviously stopping for a bit more than that, which means it takes 1000s of seconds for the computer to "cool down" again.

There's a couple of config options you can adjust to increase that limit - setting it to Integer.MAX_VALUE effectively disables this functionality entirely.

https://github.com/cc-tweaked/CC-Tweaked/blob/87a1c1a525ea97e38fbcb61253d97c7b1a6509b9/src/main/java/dan200/computercraft/shared/Config.java#L149-L159

**Reply by uecasm** [reply]
I tried changing those two settings to 2147483647, but it didn't make any difference.

Besides, it seems like it ought to recognise a silly large cooldown and just reset it.

**Reply by SquidDev** [reply]
No clue then I'm afraid. You could try stepping inside the `MainThread` class and see what's going on.

**Reply by uecasm** [reply]
Apparently pausing it for a few seconds measures a time of `45676833400`, so it still exceeded that limit.

Setting them to 1152921504606846976 instead seems to work.  (Which is not `Long.MAX_VALUE` but is close enough given that it still does maths on the value.)
